zbMATH.js

```diff
@@ -35,105 +35,163 @@
 */
 
 
-function detectWeb(doc, _url) {
-	if (ZU.xpath(doc, '//div[@class="list"]/article').length > 0) {
+function detectWeb(doc) {
+	if (doc.querySelector(".content-result")) { // search results present
 		return "multiple";
 	}
-	else if (ZU.xpath(doc, '//a[contains(@class, "bib")]').length > 0) { // contains
+	else if (doc.querySelector(".bib")) {
 		// it is a single entry --> generic fallback = journalArticle
 		return "journalArticle";
 	}
 	return false;
 }
 
-function scrape(doc, _url) {
-	var bibArray = doc.getElementsByClassName("bib");
-	var bibUrl = bibArray[0].getAttribute('href');// e.g. "bibtex/06115874.bib"
+function getSearchResults(doc, checkOnly) {
+	let items = {};
+	let found = false;
+	let rows = doc.querySelectorAll(".content-result .list > article");
+	for (let row of rows) {
+		let href = attr(row, ".title a", "href");
+		let title = ZU.trimInternal(text(row, ".title"));
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found && items;
```

Comment:
Why are you returning both here? The standard would be `	return found ? items : false;`